### PR TITLE
Disallow newlines in reason

### LINKED ISSUE #6654: Unpredictable behavior of the server with the issuance of 500 when the "reason" parameter is incorrect.
### Describe the bug

For a long time I could not understand what was the matter, when my server wrote to the logs that it returned one response with the correct status code, and on the client I received "500 Internal Server Error".

After a long discussion, I found out that the server behaves incorrectly if you specify text with several lines in the respone in the "reason" parameter, i.e. with the character "\n" inside, the server simply returns a 500 response but displays a different one in the logs.

### To Reproduce

1. make response with parameter "reason" with simbol "\n".
2. check your log on server
3. check your log on client
4. see that the client received a 500 Internal Server Error instead of the required response

### Expected behavior

It would be nice to prevent this behavior and make it explicit with a server bug!

### Logs/tracebacks

```python-traceback
No
```

### Python Version

```console
Python 3.9.1
```

### aiohttp Version

```console
Name: aiohttp
Version: 3.8.0
Summary: Async http client/server framework (asyncio)
Home-page: https://github.com/aio-libs/aiohttp
Author: 
Author-email: 
License: Apache 2
Location: /usr/local/lib/python3.9/site-packages
Requires: aiosignal, async-timeout, attrs, charset-normalizer, frozenlist, multidict, yarl
Required-by: aiobotocore, aiogram, aiohttp-swagger, core, metacore
```

### multidict Version

```console
Name: multidict
Version: 5.1.0
Summary: multidict implementation
Home-page: https://github.com/aio-libs/multidict
Author: Andrew Svetlov
Author-email: andrew.svetlov@gmail.com
License: Apache 2
Location: /home/goov/.local/lib/python3.9/site-packages
Requires: 
Required-by: aiohttp, yarl
```

### yarl Version

```console
Name: yarl
Version: 1.6.3
Summary: Yet another URL library
Home-page: https://github.com/aio-libs/yarl/
Author: Andrew Svetlov
Author-email: andrew.svetlov@gmail.com
License: Apache 2
Location: /home/goov/.local/lib/python3.9/site-packages
Requires: idna, multidict
Required-by: aio-pika, aiohttp, aiormq
```

### OS

Linux debian

### Related component

Server

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow the aio-libs Code of Conduct

### PR DESCRIPTION
**Pull request by Dreamsorcerer**
Fixes #6654.

### CODE REVIEW

File: tests/test_web_response.py [hunk 1]
```diff
@@ -1076,6 +1076,12 @@ async def test_render_with_body(buf: Any, writer: Any) -> None:
     )
 
 
+async def test_multiline_reason(buf: Any, writer: Any) -> None:
+    req = make_request("GET", "/", writer=writer)
```
Comment (Dreamsorcerer) [inline tests/test_web_response.py]:
```suggestion
```

File: aiohttp/web_exceptions.py [hunk 2]
```diff
@@ -99,6 +99,8 @@ def __init__(
         super().__init__()
         if reason is None:
             reason = self.default_reason
+        elif "\n" in reason:
+            raise ValueError("Reason cannot contain \\n")
```
Comment (webknjaz) [inline aiohttp/web_exceptions.py:103]:
```suggestion
            raise ValueError(r"Reason cannot contain \n")
```

File: aiohttp/web_response.py [hunk 3]
```diff
@@ -165,6 +165,8 @@ def set_status(
                 reason = HTTPStatus(self._status).phrase
             except ValueError:
                 reason = ""
+        if "\n" in reason:
+            raise ValueError("Reason cannot contain \\n")
```
Comment (webknjaz) [inline aiohttp/web_response.py:169]:
```suggestion
            raise ValueError(r"Reason cannot contain \n")
```

**Review by bdraco — APPROVED** [review]

File: aiohttp/web_exceptions.py (hunk 2 above)
Comment (Dreamsorcerer) [reply aiohttp/web_exceptions.py:103]: I originally felt the escape made it more explicit that this was deliberate, but can change it if you think it's more readable.

**Pull request merged**